### PR TITLE
tweak: DatePicker styling

### LINKED ISSUE #20726: [4.0] DatePicker styling problem
image

- [ ] 选中样式上月/禁用样式重叠，建议改回。
- [ ] 底部多了一个分割线。

### PR DESCRIPTION
**Pull request by ycjcl868**
Close #20726.

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

 🇺🇸English:
-DatePicker
   -💄 Optimize the border-radius of the connection and change it to rounded corners.
   -🐞 Fix selected style overlap
   -🐞 Fix extra dividing line at the bottom
   -🌟 Added `@picker-basic-cell-active-with-range-color` and `@picker-date-hover-range-color` two Less variables

🇨🇳 Chinese:

- DatePicker
  - 💄 优化连接处 border-radius，改成圆角。
  - 🐞 修复选中样式重叠
  - 🐞 修复底部额外分割线
  - 🌟 增加 `@picker-basic-cell-active-with-range-color` 和 `@picker-date-hover-range-color` 两个 Less 变量

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/date-picker/style/panel.less [hunk 1]
```diff
@@ -229,6 +229,14 @@
       background: @primary-color;
     }
 
+    &-in-view&-range-start:not(&-range-end) .@{cellClassName} {
+      border-radius: 2px 0 0 2px;
```
Comment (afc163) [inline components/date-picker/style/panel.less]: 应该用变量。

**Comment by afc163** [comment]
This PR will not close #20726.

File: components/date-picker/style/panel.less [hunk 2]
```diff
@@ -12,7 +12,6 @@
     display: inline-block;
     text-align: center;
     background: @calendar-bg;
-    border: @border-width-base @border-style-base @picker-border-color;
```
Comment (afc163) [inline components/date-picker/style/panel.less]: image

这里的边框没了。

**Comment by afc163** [comment]
多余的边框还在

图片

**Comment by ycjcl868** [comment]
@zombieJ cc, footer 没内容时 className  还在

**Comment by ycjcl868** [comment]
ci 过了就可以合了

**Pull request merged**